### ISSUE TITLE
ktlint 0.42.0 reporting incorrect indentation, and format produced bad file

### ISSUE THREAD

**Issue by ibevilinc** [pinterest/ktlint#1189]
## Expected Behavior
ktlint 0.42.0 should not report invalid indentation results

## Observed Behavior
ktlint 0.42.0 is reporting invalid indentation results.
The invalid indentation results also cause an improperly formatted file to be produced if ran with `--format`
Problem is reproducible using gradle as well as via command-line
This did not occur with version 0.41.0

## Steps to Reproduce
1. Save sample below to a file
```
fun lazyList(): Lazy<List<String>> =
    lazy { mutableListOf<String>() }

class KtlintError {

    val aListByGood:
        List<String> by lazyList()

    val aListByBad: List<String>
        by lazyList()

    val aVar = 0
}
```
2. Run ktlint from command line
```
ktlint --relative KtlintError.kt
KtlintError.kt:12:1: Unexpected indentation (4) (should be 8)
KtlintError.kt:13:1: Unexpected indentation (0) (should be 4)
KtlintError.kt:14:1: Unexpected indentation (0) (should be 4)
```

3. Running `ktlint --format` results in an incorrectly indented file (notice last variable and closing brace)
```
fun lazyList(): Lazy<List<String>> =
    lazy { mutableListOf<String>() }

class KtlintError {

    val aListByGood:
        List<String> by lazyList()

    val aListByBad: List<String>
        by lazyList()

        val aVar = 0
    }

```

## Your Environment

* Version of ktlint used: 0.42.0
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): id("org.jlleitschuh.gradle.ktlint") version "10.1.0"
* Version of Gradle used (if applicable): 7.1.1
* Operating System and version: OSX 11.5.1

**Comment by Kernald** [comment]
I also have a situation where ktlint reports an unexpected indentation of 0, should be -2 (with a 2 spaces indent size). Same, worked fine in 0.41, regressed in 0.42.

**Comment by alejandrorosas** [comment]
It happens something similar with parameters with lambdas:

Correct:
```
    AppTheme {
        Scaffold(
            topBar = {
                TopAppBar(
                    title = {
                        Text(text = "Feature Template")
                    },
                    navigationIcon = {
                        IconButton(onClick = { onNavigateBack() }) {
                            Icon(Icons.Filled.ArrowBack, contentDescription = null)
                        }
                    },
                )
            },
            floatingActionButton = {
                FloatingActionButton(onClick = { }) {
                    IconButton(onClick = { featureTemplateViewModel.onCounterClick() }) {
                        Icon(Icons.Filled.Add, contentDescription = null)
                    }
                }
            },
        ) {
            ScreenContent(counter)
        }
    }
```

Running `ktlint --format`:
```
    AppTheme {
        Scaffold(
            topBar = {
                TopAppBar(
                    title = {
                        Text(text = "Feature Template")
                    },
                    navigationIcon = {
                        IconButton(onClick = { onNavigateBack() }) {
                        Icon(Icons.Filled.ArrowBack, contentDescription = null)
                    }
                    },
                )
            },
            floatingActionButton = {
                FloatingActionButton(onClick = { }) {
                IconButton(onClick = { featureTemplateViewModel.onCounterClick() }) {
                Icon(Icons.Filled.Add, contentDescription = null)
            }
            }
            },
        ) {
            ScreenContent(counter)
        }
    }
```

**Comment by ZacSweers** [comment]
We see this issue a lot with non-trailing lambda parameters too, like RxJava subscribe sites

```kotlin
userRepositoryLazy.get().getUser(userId)
  .subscribeOn(Schedulers.io())
  .observeOn(SlackSchedulers.immediateMainThread())
  .subscribe(
    {
      viewHolder.destination.setMessagingChannelAsDestination(
        UserUtils.getDisplayName(prefsManagerLazy.get(), it),
        isDraftInThread
      )
    },
    Observers.consumerErrorLogger()
  )
```

**Comment by vRallev** [comment]
Another example in case it helps. That's what KtLint wants:
```kotlin
  @Test fun filter_returns_empty_optional_if_false() {
    val optional = Optional.of("")

    with(optional.filter { true }) {
    assertThat(this).isEqualTo(optional)
    assertThat(this.isPresent).isTrue()
  }

    with(optional.filter { false }) {
    assertThat(this).isNotEqualTo(optional)
    assertThat(this.isPresent).isFalse()
  }
  }
```

**Comment by shashachu** [comment]
I'm planning on a 0.42.1 release since this was a regression introduced to 0.42.

**Comment by ZacSweers** [comment]
I'm still seeing this issue widespread in 0.42.1, it's not just with delegated properties. Here's one example

```
    assertThat(jankyWindowsHistogram)
      .containsExactly(
        0,
        9L, // windows 4 to 11, plus implied windows 12 & 13
        60,
        2L, // windows 2 & 3
        11,
        1L, // window 1
        14,
        1L // window 0
      )
```

Which reports

```
Unexpected indentation (expected 6, actual 8)
```

It seems to happen most often with multi-line parameters to a function call that is itself already on a continuation indent

**Comment by jakoss** [comment]
I do agree with @ZacSweers , we have this in random compose code places. This does not seem to be fixed
